runtime/compiler/env/J9ClassEnv.cpp

```diff
@@ -417,24 +418,79 @@ J9::ClassEnv::isAnonymousClass(TR::Compilation *comp, TR_OpaqueClassBlock *clazz
    return comp->fej9()->isAnonymousClass(clazz);
    }
 
-const TR::TypeLayout*
-J9::ClassEnv::enumerateFields(TR::Region& region, TR_OpaqueClassBlock * opaqueClazz, TR::Compilation *comp)
-   {  
-   J9Class *clazz = (J9Class*)opaqueClazz;
-   TR_VMFieldsInfo fieldsInfo(comp, clazz, 1, stackAlloc);
-   ListIterator<TR_VMField> iter(fieldsInfo.getFields());
-   TR::TypeLayoutBuilder tlb(region);
-   for (TR_VMField *field = iter.getFirst(); field; field = iter.getNext())
+/*
+ * \param prefix
+ *    prefix could be ended with `.` or NULL.
+ * \param prefixLength
+ *    The length of the prefix string.
+ * \param toConcatenateAsPrefix
+ *    If true, the new name is appended with "." to be used as a part of the flattened field chain name.
+ * \param mergedLength
+ *    The merged length of the concatenated field name that is returned to the caller.
+ */
+static char * mergeFieldNames(char * prefix, uint32_t prefixLength, J9ROMFieldShape * field,
+                              bool toConcatenateAsPrefix, TR::Region &region, uint32_t &mergedLength)
+   {
+   J9UTF8 * fieldNameUTF = J9ROMFIELDSHAPE_NAME(field);
+   char * fieldName = reinterpret_cast<char *>(J9UTF8_DATA(fieldNameUTF));
+   uint32_t nameLength = J9UTF8_LENGTH(fieldNameUTF);
+
+   mergedLength = nameLength + prefixLength;
+   if (toConcatenateAsPrefix)
+      mergedLength++;
+   mergedLength++; /* for adding \0 at the end */
+
+   char * newName = new (region) char[mergedLength];
+
+   if (prefixLength > 0)
+      strncpy(newName, prefix, prefixLength);
+   strncpy(newName + prefixLength, fieldName, nameLength);
+
+   if (toConcatenateAsPrefix)
+      newName[mergedLength-2] = '.';
+   newName[mergedLength-1] = '\0';
+
+   return newName;
+   }
+
+static void addEntryForField(TR_VMField * field, TR::TypeLayoutBuilder &tlb, TR::Region& region, J9Class * definingClass,
+                             char * prefix, uint32_t prefixLength, IDATA offsetBase, TR::Compilation * comp)
+   {
+   J9JavaVM * vm = comp->fej9()->getJ9JITConfig()->javaVM;
+   bool bTrace = comp->getOption(TR_TraceILGen);
```

Comment:
why is this called bTrace? shouldn't it be trace or traceILGen or something a bit more intuitive?

Reply:
`b` is to indicate the variable type is a boolean. I'll change it to `trace` to be consistent with the rest of the file.